pkg/iac-providers/arm/v1/load-dir_test.go

```diff
@@ -0,0 +1,176 @@
+	"os"
+	"path/filepath"
+	"syscall"
+	"testing"
+
+	"go.uber.org/zap"
+	"gopkg.in/src-d/go-git.v4"
+	gitConfig "gopkg.in/src-d/go-git.v4/config"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+
+	"github.com/accurics/terrascan/pkg/iac-providers/output"
+	"github.com/accurics/terrascan/pkg/utils"
+	"github.com/hashicorp/go-multierror"
+)
+
+const (
+	repoURL     = "https://github.com/accurics/KaiMonkey.git"
+	branch      = "master"
+	artifacts   = "artifacts"
+	provider    = "arm"
+	testDataDir = "testdata"
+)
+
+func TestMain(m *testing.M) {
+	setup()
+	code := m.Run()
+	shutdown()
+	os.Exit(code)
+}
+
+func TestLoadIacDir(t *testing.T) {
+	invalidDirErr := &os.PathError{Err: syscall.ENOENT, Op: "lstat", Path: "not-there"}
+	if utils.IsWindowsPlatform() {
+		invalidDirErr = &os.PathError{Err: syscall.ENOENT, Op: "CreateFile", Path: "not-there"}
+	}
+
+	table := []struct {
+		name    string
+		dirPath string
+		armv1   ARMV1
+		want    output.AllResourceConfigs
+		wantErr error
+	}{
+		{
+			name:    "empty config",
+			dirPath: filepath.Join(testDataDir, "testfile"),
+			armv1:   ARMV1{},
+			wantErr: multierror.Append(fmt.Errorf("no directories found for path %s", filepath.Join(testDataDir, "testfile"))),
+		},
+		{
+			name:    "load invalid config dir",
+			dirPath: testDataDir,
+			armv1:   ARMV1{},
+			wantErr: nil,
+		},
+		{
+			name:    "invalid dirPath",
+			dirPath: "not-there",
+			armv1:   ARMV1{},
+			wantErr: multierror.Append(invalidDirErr),
+		},
+		{
+			name:    "key-vault",
+			dirPath: filepath.Join(testDataDir, "key-vault"),
+			armv1:   ARMV1{},
+			wantErr: nil,
+		},
+	}
+
+	for _, tt := range table {
+		t.Run(tt.name, func(t *testing.T) {
+			_, gotErr := tt.armv1.LoadIacDir(tt.dirPath, false)
+			me, ok := gotErr.(*multierror.Error)
+			if !ok {
+				t.Errorf("expected multierror.Error, got %T", gotErr)
+			}
+			if tt.wantErr == nil {
+				if err := me.ErrorOrNil(); err != nil {
+					t.Errorf("unexpected error; gotErr: '%v', wantErr: '%v'", gotErr, tt.wantErr)
+				}
+			} else if me.Error() != tt.wantErr.Error() {
+				t.Errorf("unexpected error; gotErr: '%v', wantErr: '%v'", gotErr, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestARMMapper(t *testing.T) {
+	root := filepath.Join(artifacts, provider)
+	dirList := make([]string, 0)
+	err := filepath.Walk(root, func(filePath string, fileInfo os.FileInfo, err error) error {
+		if fileInfo != nil && fileInfo.IsDir() {
+			dirList = append(dirList, filePath)
+		}
+		return err
+	})
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	armv1 := ARMV1{}
+	for i := 1; i < len(dirList); i++ {
+		dir := dirList[i]
+		t.Run(dir, func(t *testing.T) {
+			_, gotErr := armv1.LoadIacDir(dir, false)
+			_, ok := gotErr.(*multierror.Error)
+			if !ok {
+				t.Errorf("expected multierror.Error, got %T", gotErr)
+			}
+		})
+	}
+}
+
+func setup() {
+	err := downloadArtifacts()
+	if err != nil {
+		zap.S().Fatal(err)
+	}
+}
+
+func shutdown() {
+	os.RemoveAll(artifacts)
+}
+
+func downloadArtifacts() error {
```

Comment:
Can we see if the `downloder` package available in terrascan can be used here?

Reply:
I tried first with downloader package but it checks for source url and gives this error: https://github.com/hashicorp/go-getter/blob/ad4c48e1063c1794702c37900d827a645bc9a79e/get_http.go#L122

Reply:
Seems like downloader is specific to terraform remote modules